src/platform/i386/vm.c

```diff
@@ -17,6 +17,9 @@ struct liveness_entry __liveness_tbl[LTBL_ENTS] CACHE_ALIGNED LARGE_BSS;
 u32_t boot_comp_pgd[PAGE_SIZE / sizeof(u32_t)] PAGE_ALIGNED = {[0] = 0 | PGTBL_PRESENT | PGTBL_WRITABLE | PGTBL_SUPER,
                                                                [KERN_INIT_PGD_IDX] = 0 | PGTBL_PRESENT | PGTBL_WRITABLE
                                                                                      | PGTBL_SUPER};
+u32_t boot_ap_pgd[PAGE_SIZE / sizeof(u32_t)] PAGE_ALIGNED = {[0] = 0 | PGTBL_PRESENT | PGTBL_WRITABLE | PGTBL_SUPER,
```

Comment:
Is there a reason to not make `boot_comp_pgd` a 2D array with NUM_CPU? I feel like that would be better.

Reply:
Wait.. I thought we share the pgtbls across cores. 
so we don't need a 2D array or a `boot_ap_pgd`, right?

Reply:
I forget the core of this issue.  @yzcode ?

Reply:
We cannot reuse the `boot_comp_pgd` for APs' booting because we will unmap the lower address of `boot_comp_pgd` in `kern_setup_image` later. But we will switch the AP's pgd back to `boot_comp_pgd` after the AP is booted into C-level to keep the simplicity.

Reply:
What is the usual way of multi-core booting? Does the BSP continue in user-land while the other processors are still booting? 

If that is not the case, we probably can delay the call to `kern_setup_image` (and perhaps subsequent ones) in boot processor for until all other APs get to that point? 

What you've now maybe OK but we need to we very sure we don't have a mapping boot_comp_pgd that's used by other APs or we don't have a mapping in AP that's not in boot_comp_pgd.

Reply:
@phanikishoreg in my understanding, the boot_ap_pgd will only be used in the assembly booting code, all cores will share the boot_comp_pgd after that.